src-electron/db/query-attribute.js

```diff
@@ -138,6 +138,26 @@ async function duplicateEndpointTypeAttribute(
   )
 }
 
+/**
+ * Promises to delete endpoint type attributes by their endpoint-type-ref and cluster-ref.
+ *
+ * @export
+ * @param {*} db
+ * @param {*} endpointTypeRef
+ * @returns Promise to delete endpoint type.
+ */
+async function deleteEndpointTypeAttributes(
```

Comment:
Do not do delete attributes from the endpoint_type_attribute table. If you do this then when the user enables the same cluster then they loose their old selections.
What you want to do instead is check if the cluster is enabled or not and then count the attributes.